tests/Doctrine/Tests/Persistence/Mapping/ClassMetadataFactoryTest.php

```diff
@@ -34,6 +35,17 @@ public function testGetCacheDriver()
         self::assertSame($cache, $this->cmf->getCacheDriver());
     }
 
+    public function testGetMetadataForAnonymousClass()
+    {
+        $this->expectException(MappingException::class);
+        $this->expectExceptionMessage('anonymous');
+
+        $object = new class () extends stdClass {
+        };
+
+        $this->cmf->getMetadataFor(get_class($object));
```

Comment:
It's a good practice to always put the exception expectations right before the call that triggers them.

```suggestion
        $object = new class () extends stdClass {
        };

        $this->expectException(MappingException::class);
        $this->expectExceptionMessage('anonymous');
        
        $this->cmf->getMetadataFor(get_class($object));
```